.github/workflows/main.yml

```diff
@@ -81,3 +81,62 @@ jobs:
         with:
           fetch-depth: 0
       - uses: ./tools/github-actions/cascading
+  documentation:
+    runs-on: ubuntu-latest
+    needs: [version, build, checks]
+    if: ${{ github.ref_name == 'main' || github.base_ref == 'main' }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./tools/github-actions/setup
+      - name: Generate doc
+        run: yarn doc:generate
+      - name: Generate package json
+        working-directory: ./generated-doc
+        run: yarn init -y
+      - name: Update package json
+        run: yarn doc:update-package ./generated-doc/package.json --name @o3r/framework-doc --version ${{ needs.version.outputs.nextVersionTag }}
+      - uses: actions/upload-artifact@v3
+        with:
+          name: documentation
+          path: ./generated-doc
+      - uses: actions/upload-pages-artifact@v1
+        with:
+          path: ./generated-doc
+  deploy-documentation-azure-static-webapp:
+    runs-on: ubuntu-latest
+    needs: [documentation]
+    environment:
+      name: azure-static-webapp
+      url: ${{ steps.deploy.outputs.static_web_app_url }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: documentation
+          path: './generated-doc'
+      - name: deploy
+        id: deploy
+        uses: Azure/static-web-apps-deploy@v1
+        with:
+          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN }}
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          action: upload
+          app_location: ./generated-doc
+          skip_app_build: true
+          skip_api_build: true
+          config_file_location: '.'
+  deploy-documentation-github-pages:
```

Comment:
This is a bit sad to deploy the documentation to the unique page representing the framework on github.
Maybe we should keep this page to a more fancy portal?